File: backend/src/main/scala/microservices/operations/airline/objects/RegisterAirlineManagerPlannerRequest.scala
// 这个文件只定义 airline 管理员注册请求。
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class RegisterAirlineManagerPlannerRequest(email: String, displayName: String, airlineName: String, airlineCode: String, password: String)
object RegisterAirlineManagerPlannerRequest:
  given sourceEncoder: Encoder[RegisterAirlineManagerPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[RegisterAirlineManagerPlannerRequest] = deriveDecoder

File: frontend/src/microservices/operations/airline/objects/RegisterAirlineManagerPlannerRequest.ts
// 本文件定义 RegisterAirlineManagerPlanner 的航空管理员注册请求。

export type RegisterAirlineManagerPlannerRequest = {
  email: string
  displayName: string
  airlineName: string
  airlineCode: string
  password: string
}

export const registerAirlineManagerPlannerRequestFromJson = (json: string): RegisterAirlineManagerPlannerRequest =>
  JSON.parse(json) as RegisterAirlineManagerPlannerRequest

export const registerAirlineManagerPlannerRequestToJson = (value: RegisterAirlineManagerPlannerRequest): string =>
  JSON.stringify(value)
